### PR TITLE
Engine spring cleaning

### PR DESCRIPTION
**Pull request by sim0504**
Just look at the code. Comment if you see stupid stuffs. Not ready to test yet!

Also, I found a small bug in the Kalman `dt` calculation.

### CODE REVIEW

File: Engine/controller.py [hunk 1]
```diff
@@ -5,111 +5,103 @@
...
-            self.ui_send_queue.put_nowait(DebugCommandFactory.plot_point("mm/s",
-                                                                         "robot {} kallman speed".format(robot.robot_id),
-                                                                         [time.time()],
-                                                                         [robot.velocity.norm]))
-
-            commands[robot.robot_id] = self._put_in_robots_referential(robot, cmd)
+
+        for robot in self.active_robots:
+            robot.path, robot.target_speed = path_smoother(robot.raw_path, robot.cruise_speed, robot.end_speed)
+            commands[robot.id] = robot.velocity_regulator.execute(robot)
```
Comment (sim0504) [inline Engine/controller.py]: I remove the position regulator according to a comment from @wonwon0 . Could be readded if needed.

**Comment by sim0504** [comment]
J'ai testé quelques strategies, semblent bien fonctionner.

File: Engine/tracker.py [hunk 2]
```diff
@@ -5,209 +5,203 @@
...
+    def _change_frame_side(detection_frame: Dict[str, List[Dict[str, Any]]]) -> Dict[str, List[Dict[str, Any]]]:
 
-        teams = detection_frame.get('robots_blue', []) \
-              + detection_frame.get('robots_yellow', [])
-
-        for robot_obs in teams:
+        for robot_obs in detection_frame.get('robots_blue', []) + detection_frame.get('robots_yellow', []):
             robot_obs['x'] *= -1
-            robot_obs['orientation'] = wrap_to_pi(np.pi - robot_obs['orientation'])
+            robot_obs['orientation'] *= -1
```
Comment (PhiBabin) [inline Engine/tracker.py]: We need to mirror the robot orientation around the y axis. Not around the x axis. For example a robot with a 110 deg orientation has for mirror 70 deg, not -110 deg

Comment (sim0504) [reply Engine/tracker.py]: oh, i copied pasted the other computation...

Comment (sim0504) [reply Engine/tracker.py]: oh... ust realize it's not the same thing at all lawl sorry

**Pull request merged**